python/cudf/cudf/core/column/string.py

```diff
@@ -533,6 +535,46 @@ def isnumeric(self, **kwargs):
         """
         return self._return_or_inplace(cpp_is_numeric(self._column))
 
+    def isupper(self, **kwargs):
+        """
+        Returns a Series/Index of boolean values with True for strings
+        that contain only upper-case characters.
+
+        Returns
+        -------
+        Series/Index of bool dtype
+
+        Notes
+        -----
+        Results are incompatible with standard python string logic. Use caution
+        when operating on data which contains non-alphabetical characters.
+        """
+        warnings.warn(
+            "isupper does not conform to standard python string logic"
+        )
```

Comment:
```suggestion
        warnings.warn(
            "isupper currently returns False for non-cased characters whereas Pandas returns True, this will be fixed in the near future"
        )
```